### PR TITLE
Add prom command line args and validation

### LINKED ISSUE #52: Change Prometheus port
Hi!

Is there any option to change the Prometheus port to something different than 9090? I cannot see such options anywhere in the code, but maybe it is buried somewhere. I can make PR and add this possibility to specify a port, it this is not a problem.

### PR DESCRIPTION
**Pull request by hooten**
#### Pull Request (PR) description
This pull request adds the remaining command-line arguments available to Prometheus v2 (and a few for v1). It also adds validation that will fail the puppet run if arguments are used incorrectly. It continues to expose the `extra_options` parameter for custom command-line logic. It also adds a number of tests for using command line arguments.

Although it was possible to specify arbitrary command-line arguments previously using `extra_options`, this PR makes CLI args more discoverable and manages better.

#### This Pull Request (PR) fixes the following issues
Fixes #52

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @hooten, thanks for the PR. Can you take a look at the failing travis tests?

File: manifests/init.pp [hunk 1]
```diff
@@ -198,13 +340,45 @@
   Optional[String[1]] $extract_command,
   Boolean $manage_config,
   Stdlib::Absolutepath $usershell,
-  Hash $extra_alerts    = {},
-  Hash $config_hash     = {},
-  Hash $config_defaults = {},
-  String $os            = downcase($facts['kernel']),
+  Optional[String] $web_listen_address                                          = undef,
```
Comment (bastelfreak) [inline manifests/init.pp]: for those datatypes, please ensure empty strings are prohibited:
```suggestion
  Optional[String[1]] $web_listen_address                                       = undef,
```

File: manifests/init.pp [hunk 2]
```diff
@@ -198,13 +340,45 @@
...
+  Optional[String] $web_read_timeout                                            = undef,
+  Optional[String] $web_max_connections                                         = undef,
+  Optional[String] $web_route_prefix                                            = undef,
+  Optional[String] $web_user_assets                                             = undef,
+  Optional[Boolean] $web_enable_lifecycle                                       = undef,
```
Comment (bastelfreak) [inline manifests/init.pp]: for those boolean parameters, can you remove the `Optional[]` and set the default to true/false?

**Comment by hooten** [comment]
Thanks for the commends @bastelfreak. This PR is still a WIP. I'm having trouble reproducing the failing tests locally. I will work on it more this week

File: manifests/init.pp (hunk 2 above)
Comment (hooten) [reply manifests/init.pp]: can do!

File: manifests/init.pp (hunk 1 above)
Comment (hooten) [reply manifests/init.pp]: sounds good!

**Comment by hooten** [comment]
Hey @bastelfreak, I made the changes you suggested and fixed the tests. When you get a chance, I'd really appreciate another review. Thanks!

**Comment by bastelfreak** [comment]
Thanks for the work!

**Comment by hooten** [comment]
Thanks! I'm happy to contribute!

**Pull request merged**